### PR TITLE
Add Tesla middleware

### PR DESCRIPTION
**Pull request by ricardoccpaiva**
This Tesla middleware injects tracing headers into HTTP requests so that we get distributed tracing for free.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,24 +26,19 @@ by adding `opentelemetry_tesla` to your list of dependencies in `mix.exs`:
...
-  ]
-  opts = [strategy: :one_for_one, name: MyApp.Supervisor]
-  Supervisor.start_link(children, opts)
-end
+OpenTelemetry.register_application_tracer(:my_telemetry_api)
+OpentelemetryTesla.setup()
 ```
 
-After this, spans will start to be created whenever a request is completed or if it eventually fails with an exception.
+On the other hand, if you want to use the middleware, just add it like any other Tesla middleware.
```
Comment (indrekj) [inline README.md:41]: I would say something in lines of `To also propagate the tracing information you'll also have to add the tesla middleware:`. Just to be more clear that combining these will do both the span creation and also propagation.

File: lib/middleware/opentelemetry_tesla_middleware.ex [hunk 2]
```diff
@@ -0,0 +1,9 @@
+defmodule Tesla.Middleware.OpenTelemetry do
```
Comment (indrekj) [inline lib/middleware/opentelemetry_tesla_middleware.ex:1]: I wonder if this should be something like OpentelemetryTesla.Middleware to indicate it's from this package. Not sure about the convention here.

File: mix.exs [hunk 3]
```diff
@@ -4,7 +4,7 @@ defmodule OpentelemetryTesla.MixProject do
   def project do
     [
       app: :opentelemetry_tesla,
-      version: "1.0.0-rc.1",
+      version: "1.1.0-rc.1",
```
Comment (indrekj) [inline mix.exs:7]: as it was not stable and it's not backwards-incompatible change, I'd go with 0.2.0-rc.1

**Review by indrekj — APPROVED** [review]

File: lib/middleware/opentelemetry_tesla_middleware.ex (hunk 2 above)
Comment (ricardoccpaiva) [reply lib/middleware/opentelemetry_tesla_middleware.ex:1]: As far as I can see [here](https://github.com/teamon/tesla#custom-middleware), it's ok.
I've looked for a few middlewares online and the naming strategy is not always the same, some use the one like this PR and other ones are completely custom "à lá carte".

File: mix.exs (hunk 3 above)
Comment (ricardoccpaiva) [reply mix.exs:7]: Actually, I think I didn't nailed it from the beginning. Even though I agree with your comment isn't a bit weird going from a `1.0.0` to a `0.2.0`?

Comment (indrekj) [reply mix.exs:7]: oh nvm. I thought the previous version was 0.1. Going from 1.0 to 1.1 is totally fine here.

**Pull request merged**